Fix Unable to fetch logs from worker pod error in UI for k8s executor

snjypl opened the pull request:
_[ note: i have set `delete_worker_pods` to `False` so that the pod is not deleted after completion]_

for k8s executor while trying to view the task log from the UI.  we are getting the following error. 

```
airflow-webserver-5bf48475c-zdjxv
*** Trying to get logs (last 100 lines) from worker pod airflow-webserver-5bf48475c-zdjxv ***

*** Unable to fetch logs from worker pod airflow-webserver-5bf48475c-zdjxv ***
('Cannot find pod for ti %s', <TaskInstance: dataset_produces_2.producing_task_2 manual__2022-12-28T21:13:27.229615+00:00 [success]>)
```

i think, the issue is, while calling `PodGenerator.build_selector_for_k8s_executor_pod` we are passing `ti.try_number` instead of the `try_number` that was passed to the `_read` method. 

https://github.com/apache/airflow/blob/3ececb2c79307bd943aad116d7b0b5933af8185a/airflow/utils/log/file_task_handler.py#L167

https://github.com/apache/airflow/blob/3ececb2c79307bd943aad116d7b0b5933af8185a/airflow/utils/log/file_task_handler.py#L215-L222



<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

o-nikolas reviewed (approved):
lgtm, pending passing tests of course.

dstandish commented on airflow/executors/base_executor.py:
```diff
@@ -356,13 +356,13 @@ def execute_async(
         raise NotImplementedError()
 
-    def get_task_log(self, ti: TaskInstance) -> tuple[list[str], list[str]]:
+    def get_task_log(self, ti: TaskInstance, try_number: int) -> tuple[list[str], list[str]]:
         """
```
wait this bit doesn't make sense to me.  if we already have the TI, why is it we also need to supply the try number?

o-nikolas replied:
There is some explanation in the code for `read`: https://github.com/apache/airflow/blob/c266aacf39a2f01cb3f91740f448934c7b3fd7be/airflow/utils/log/file_task_handler.py#L376-L410

`try_number` is plumbed in from `read` to `_read` and then should have been sent to the executors, but it was a miss:
https://github.com/apache/airflow/blob/c266aacf39a2f01cb3f91740f448934c7b3fd7be/airflow/utils/log/file_task_handler.py#L274-L315

snjypl committed:
try_number was not being passed to the get_task_log method, instead
ti.try_number was used for fetching log from k8s pod.
it was causing incorrect log being returned for k8s pod.
fixed by passing try_number from _read to get_task_log method
use try_number argument instead of ti.try_number for selecting pod in
k8s executor

eladkal committed:
Merge branch 'main' into bugfix/Fix-Unable-to-fetch-logs-from-worker-pod-error-in-UI-for-k8s-executor
